mldsa/src/poly.c

```diff
@@ -237,16 +264,7 @@ void mld_poly_pointwise_montgomery(mld_poly *c, const mld_poly *a,
     }
   }
 #endif /* MLD_USE_NATIVE_POINTWISE_MONTGOMERY */
-  for (i = 0; i < MLDSA_N; ++i)
-  __loop__(
-    invariant(i <= MLDSA_N)
-    invariant(array_abs_bound(c->coeffs, 0, i, MLDSA_Q))
-  )
-  {
-    c->coeffs[i] = mld_montgomery_reduce((int64_t)a->coeffs[i] * b->coeffs[i]);
-  }
-
-  mld_assert_abs_bound(c->coeffs, MLDSA_N, MLDSA_Q);
+  mld_poly_pointwise_montgomery_c(c, a, b);
```

Comment:
The use of assertions is not consistent here. We should either a) use pre/post assertions _everywhere_ -- in which case we should have a post-condition assertion here as well, or b) we allow ourselves -- for test performance -- to omit 'obviously' redundant pre/post conditions, such as when we call from `foo` straight into `foo_c`.

WDYT @mkannwischer @willieyz

Reply:
I'm saying the assertions are inconsistent. We have a pre-condition assert at the toplevel, but no post-condition check. Indeed the pre-condition check could be moved into the native branch since it, too, is duplicated with the precondition assert in `foo_c`. If one wanted to be pedantic and consistent, though, and _always_ have pre/post asserts matching a function's contract, I'd be OK with duplicated post-conditions as well.

Reply:
Okay, let's move the pre-condition into the native branch as proposed in the other comment.